mongoimport/mongoimport_test.go

```diff
@@ -754,6 +765,43 @@ func TestImportDocuments(t *testing.T) {
 			}
 			So(checkOnlyHasDocuments(*imp.SessionProvider, expectedDocuments), ShouldBeNil)
 		})
+		Convey("CSV import with --mode=delete should succeed", func() {
+			// Fist import 3 documetns
+			imp, err := NewMongoImport()
+			So(err, ShouldBeNil)
+
+			imp.InputOptions.Type = CSV
+			imp.InputOptions.File = "testdata/test.csv"
+			fields := "_id,c,b"
+			imp.InputOptions.Fields = &fields
+			imp.IngestOptions.MaintainInsertionOrder = true
+			numImported, numFailed, err := imp.ImportDocuments()
+			So(err, ShouldBeNil)
+			So(numImported, ShouldEqual, 3)
+			So(numFailed, ShouldEqual, 0)
+
+			// Then delete two documents
+			imp, err = NewMongoImport()
+			So(err, ShouldBeNil)
+
+			imp.InputOptions.Type = CSV
+			imp.InputOptions.File = "testdata/test_delete.csv"
+			fields = "_id,c,b"
+			imp.InputOptions.Fields = &fields
+			imp.IngestOptions.Mode = modeDelete
+			imp.IngestOptions.StopOnError = true
+			// Must specify upsert fields since option parsing is skipped in tests
+			imp.upsertFields = []string{"_id"}
```

Comment:
Should add a test that uses a non-ID field to delete -- which gives me an interesting question in that case... are we doing "delete one" or are we doing "delete many"?  To be consistent with other modes, I think we should be intending this to be single-document -- and so we need to document that the update fields should uniquely identify a document.

Reply:
Yes it is "delete one".